src/modules/schema.js

```diff
@@ -0,0 +1,189 @@
+} from './blob';
+
+// Supported schemas
+import JSONSchema07 from '../utils/vc/schemas/schema-draft-07';
+
+export default class Schema {
+  /**
+   * Creates a new `Schema` object
+   * @constructor
+   * @param {string} [id] - optional schema ID, if not given, generate a random id
+   */
+  constructor(id) {
+    this.id = id || createNewDockBlobId();
+    this.name = '';
+    this.version = '1.0.0';
+  }
+
+  /**
+   * Add the JSON schema to this object after checking that `json` is a valid JSON schema. Check if JSON is valid.
+   * @param {object} json - the schema JSON
+   */
+  async setJSONSchema(json) {
+    await Schema.validateSchema(json);
+    this.schema = json;
+    return this;
+  }
+
+  setName(name) {
+    this.name = name;
+    return this;
+  }
+
+  /**
+   * Update the object with `author` key. Repeatedly calling it will keep resetting the author
+   * did can be a DID hex identifier or full DID
+   * @param {string} did - the author DID
+   */
+  setAuthor(did) {
+    if (did.startsWith('did:') || isHexWithGivenByteSize(did, DockBlobIdByteSize)) {
+      this.author = did;
+    } else {
+      throw new Error(`Supplied author ${did} is not a valid DID or ${DockBlobIdByteSize} byte hex string`);
+    }
+    return this;
+  }
+
+  /**
+   * Update the object with `signature` key. This method is used when
+   * signing key/capability is not present and the signature is received from outside.
+   * Repeatedly calling it will keep resetting the `signature` key.
+   * The signature must be one of the supported objects
+   * @param {object} signature - The schema's signature
+   */
+  setSignature(signature) {
+    if (signature instanceof Signature) {
+      this.signature = signature;
+    } else {
+      throw new Error('Provided signature object is not of instance Signature');
+    }
+
+    return this;
+  }
+
+  /**
+   * Serializes the object using `getSerializedBlob` and then signs it using the given
+   * polkadot-js pair. The object will be updated with key `signature`. Repeatedly calling it will
+   * keep resetting the `signature` key
+   * @param {any} msg - The message to sign
+   * @param {object} pair - Key pair to sign with
+   */
+  sign(msg, pair) {
+    this.signature = getSignatureFromKeyringPair(pair, msg);
+    return this;
+  }
+
+  /**
+   * Serializes to JSON for sending to the node, as `Blob`. A full DID is converted to the
+   * hex identifier and signature object is converted to the enum
+   * @returns {object}
+   */
+  toJSON() {
+    const {
+      signature,
+      id,
+      ...rest
+    } = this;
+
+    return {
+      ...rest,
+      id: getHexIdentifierFromBlobID(this.id),
+      signature: signature.toJSON(),
+    };
+  }
+
+  /**
+   * Serializes the schema to a blob object to send to the node
+   * @returns {object}
+   */
+  toBlob(author) {
+    return {
+      id: this.id,
+      blob: stringToHex(JSON.stringify(this.toJSON())),
+      author: getHexIdentifierFromDID(author),
```

Comment:
Any reason you want `this.author` specifically to be overwritable? If not then you can use `this.author`

Reply:
overlooked this during implementation of toblob method, good spot will adjust